### PR TITLE
fix(aws-ecs): ensure cluster attributes are accessible from constructor’s props

### PR DESCRIPTION
**Pull request by piradeepk**
Ensuring all settings are accessible from the cluster constructor’s props argument (ex: properties in addDefaultCloudMapNamespace method, and addCapacity method.

----

Please read the [contribution guidelines](https://github.com/awslabs/aws-cdk/blob/master/CONTRIBUTING.md) and follow the pull-request checklist.

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

----

### CODE REVIEW

**Comment by pkandasamy91** [comment]
@rix0rrr @eladb please review

File: packages/@aws-cdk/aws-ecs/lib/cluster.ts [hunk 1]
```diff
@@ -23,6 +23,27 @@ export interface ClusterProps {
...
+   * @default none
+   */
+  readonly defaultCloudMapNamespace?: NamespaceOptions;
+
+  /**
+   * The name of the autoscaling group to be added
+   *
+   * @default none
+   */
+  readonly autoscalingGroupName?: string;
```
Comment (eladb) [inline packages/@aws-cdk/aws-ecs/lib/cluster.ts]: Remove this prop, I don't think this is needed, we can just use something like `"Default"` as the default name.

File: packages/@aws-cdk/aws-ecs/lib/cluster.ts [hunk 2]
```diff
@@ -23,6 +23,27 @@ export interface ClusterProps {
...
+
+  /**
+   * The ec2 capacity to add to the cluster
+   *
+   * @default none
```
Comment (eladb) [inline packages/@aws-cdk/aws-ecs/lib/cluster.ts]: Use `-` as a value and describe the default behavior and reference the method:

```ts
@default - no EC2 capacity will be added, you can use `addCapacity` to add capacity later.
```

File: packages/@aws-cdk/aws-ecs/lib/cluster.ts [hunk 3]
```diff
@@ -23,6 +23,27 @@ export interface ClusterProps {
    * The VPC where your ECS instances will be running or your ENIs will be deployed
    */
   readonly vpc: ec2.IVpc;
+
+  /**
+   * The service discovery namespace created in this cluster
+   *
+   * @default none
```
Comment (eladb) [inline packages/@aws-cdk/aws-ecs/lib/cluster.ts]: Use `-` as a default value and describe the behavior. Reference the method

File: packages/@aws-cdk/aws-ecs/lib/cluster.ts [hunk 4]
```diff
@@ -23,6 +23,27 @@ export interface ClusterProps {
...
+   */
+  readonly defaultCloudMapNamespace?: NamespaceOptions;
```
Comment (eladb) [inline packages/@aws-cdk/aws-ecs/lib/cluster.ts]: Shouldn't that have been `CloudMapNamespaceOptions`?

File: packages/@aws-cdk/aws-ecs/lib/cluster.ts [hunk 5]
```diff
@@ -365,6 +411,13 @@ export interface ClusterAttributes {
    * @default - No default namespace
    */
   readonly defaultCloudMapNamespace?: cloudmap.INamespace;
+
+  /**
+   * Getter for autoscaling group added to the cluster
```
Comment (eladb) [inline packages/@aws-cdk/aws-ecs/lib/cluster.ts]: "Getter"?

File: packages/@aws-cdk/aws-ecs/lib/cluster.ts (hunk 2 above)
Comment (pkandasamy91) [reply packages/@aws-cdk/aws-ecs/lib/cluster.ts]: Addressed.

File: packages/@aws-cdk/aws-ecs/lib/cluster.ts (hunk 3 above)
Comment (pkandasamy91) [reply packages/@aws-cdk/aws-ecs/lib/cluster.ts]: Addressed.

File: packages/@aws-cdk/aws-ecs/lib/cluster.ts (hunk 4 above)
Comment (pkandasamy91) [reply packages/@aws-cdk/aws-ecs/lib/cluster.ts]: Renaming NamespaceOptions was a separate PR that was merged in. Addressed.

File: packages/@aws-cdk/aws-ecs/lib/cluster.ts (hunk 5 above)
Comment (pkandasamy91) [reply packages/@aws-cdk/aws-ecs/lib/cluster.ts]: If the autoscaling group is created in the constructor, there needs to be a way to get it. (The AddCapacity method returns the ASG)

File: packages/@aws-cdk/aws-ecs/lib/cluster.ts (hunk 1 above)
Comment (pkandasamy91) [reply packages/@aws-cdk/aws-ecs/lib/cluster.ts]: Removed and defaulted to 'DefaultAutoScalingGroup'

**Comment by pkandasamy91** [comment]
Renamed the PR

File: packages/@aws-cdk/aws-ecs/lib/cluster.ts (hunk 5 above)
Comment (pkandasamy91) [reply packages/@aws-cdk/aws-ecs/lib/cluster.ts]: Updated all comments to remove Getter terminology

**Comment by pkandasamy91** [comment]
Not sure why the commit-lint-pr keeps hanging - looks like it's happening on multiple PRs. Can this be merged without it completing?

**Comment by pkandasamy91** [comment]
Rebased and resolved merge conflicts

**Comment by pkandasamy91** [comment]
@rix0rrr @eladb please merge

**Comment by pkandasamy91** [comment]
@eladb can this be merged in?

**Pull request merged**